src/App.tsx

```diff
@@ -16,47 +18,93 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const [visibleGoods, setVisibleGoods] = useState<string[]>([]);
+  const [sortType, setSortType] = useState<SortType>(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+
+  useEffect(() => {
+    const goodsForShow = [...goodsFromServer];
+
+    switch (sortType) {
+      case SortType.ALPHABET:
+        goodsForShow.sort((a, b) => a.localeCompare(b));
+        break;
+
+      case SortType.LENGTH:
+        goodsForShow.sort((a, b) => a.length - b.length);
+        break;
+
+      default:
+    }
+
+    if (isReversed) {
+      goodsForShow.reverse();
+    }
+
+    setVisibleGoods(goodsForShow);
+  }, [sortType, isReversed]);
```

Comment:
It's a good solution, but you could combine `useEffect` with useState.
[useMemo](https://reactjs.org/docs/hooks-reference.html#usememo)
But, you will learn about it later